Fixed some crashes, refactoring

calvinnor opened the pull request:
Issue:
- When the user tries to rotate a GIF, the app crashes
- When the user taps on Media to hide / show SystemUI,
  the app crashes

Root Cause:
- Rotating Gifs is not a supported feature
- When this action is tried, the code fetches the current fragment
  and casts it into an ImageFragment, which it's not
- We call toggleSystemUI on the Activity, which can be null.

Fixes:
- Added a state boolean to maintain if we are showing the Image
  Menu (rotate, use as) or not.
- We now only invalidate the menu on ViewPager swipe if the new
  media needs a different Menu (image - video - gif)
- Added BaseMediaFragment to hold boilerplate code of getting the
  Media, and toggling System UI.
- Removed all Activity dependencies in Media Fragments.
- Minor cleanup and fixes.

dnldsht commented on app/src/main/java/org/horaapps/leafpic/activities/MainActivity.java:
```diff
@@ -168,21 +168,8 @@ public void displayMedia(Album album) {
-                intent.putExtra("position", position);
-                startActivity(intent);
-            } catch (Exception e) {
-                intent.setAction(SingleMediaActivity.ACTION_OPEN_ALBUM_LAZY);
```
this is necessary when there are a lot of media in an album, the Bundle has a size limit.

calvinnor committed:
Issue:
- When the user tries to rotate a GIF, the app crashes
- When the user taps on Media to hide / show SystemUI,
  the app crashes

Root Cause:
- Rotating Gifs is not a supported feature
- When this action is tried, the code fetches the current fragment
  and casts it into an ImageFragment, which it's not
- We call toggleSystemUI on the Activity, which can be null.

Fixes:
- Added a state boolean to maintain if we are showing the Image
  Menu (rotate, use as) or not.
- We now only invalidate the menu on ViewPager swipe if the new
  media needs a different Menu (image - video - gif)
- Added BaseMediaFragment to hold boilerplate code of getting the
  Media, and toggling System UI.
- Removed all Activity dependencies in Media Fragments.
- Minor cleanup and fixes.

CalvinNor commented:
@dnldsht Reverted the required changes for TransactionTooLarge exception being thrown.
This PR is good to go.
